packages/core/scripts/local/EncodeParams.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @notice The purpose of this script is to encode ExpiringMultiParty parameters.
+ *
+ * Spinning up mainnet fork: ganache-cli --fork https://mainnet.infura.io/v3/5f56f0a4c8844c96a430fbd3d7993e39 --unlock 0x2bAaA41d155ad8a4126184950B31F50A1513cE25 --unlock 0x7a3a1c2de64f20eb5e916f40d11b01c441b2a8dc --port 9545
+ * Example: yarn truffle exec ./packages/core/scripts/local/EncodeParams.js --cversion latest --network mainnet-fork
+ */
+const { toWei } = web3.utils;
+const { ZERO_ADDRESS } = require("@uma/common");
+const { getAbi, getTruffleContract } = require("../../index");
+const argv = require("minimist")(process.argv.slice(), {
+  string: ["cversion"]
+});
+const abiVersion = argv.cversion || "latest"; // Default to most recent mainnet deployment, 1.2.2.
+
+// Deployed contract ABI's and addresses we need to fetch.
+const ExpiringMultiPartyCreator = getTruffleContract("ExpiringMultiPartyCreator", web3, abiVersion);
+const ExpiringMultiParty = getTruffleContract("ExpiringMultiParty", web3, abiVersion);
+const Finder = getTruffleContract("Finder", web3, abiVersion);
+const TokenFactory = getTruffleContract("TokenFactory", web3, abiVersion);
+
+/** ***************************************************
+ * Main Script
+ /*****************************************************/
+const encodeParams = async callback => {
+  try {
+    const expiringMultiPartyCreator = await ExpiringMultiPartyCreator.deployed();
+    const finder = await Finder.deployed();
+    const tokenFactory = await TokenFactory.deployed();
+
+    // Replace these with the EMP constructor params you want to encode
+    let constructorParams = {
```

Comment:
Could we decode these from the on-chain transaction?

Something like this, maybe:

```js
// At the top
const abiDecoder = require("abi-decoder");

...
// this could also just be a single abi if you know which one you want.
const abis = [getAbi("ExpiringMultiParty", "1.1.0"), getAbi("ExpiringMultiParty", "1.2.0") ....];
abis.forEach(abi => abiDecoder.addAbi(abi));
let constructorParams = abiDecoder.decode(transactionData).params;
```

This may be a little annoying because of some of the versioning below, but just wanted to mention it as a possibility. Not saying we necessarily _should_ do this.

Reply:
Hmm, I think we probably could. I'll try rolling this into a follow-up PR when we encode params for the perpetual.